ntrans_gui.py

```diff
@@ -0,0 +1,293 @@
+import webbrowser
+import tkinter
+from tkinter import filedialog
+from tkinter import ttk
+
+
+class NTransMainGui:
+    def __init__(self):
+        # User-determined variables
+        self.included_ngrams = [2, 3, 4, 5, 6]
+
+        # Window & Frame
+        self.root = tkinter.Tk()
+        self.root.resizable(False, False)
+        self.root.title("N-Trans")
+
+        self.mainframe = ttk.Frame(self.root)
+        self.mainframe.pack(fill="both", expand=True)
+
+        # Header
+        header = ttk.Label(
+            self.mainframe,
+            text="N-Trans Dictionary Settings",
+            font=("TkDefaultFont", 18),
+        )
+
+        # Get directory path to save csv file
+        # TODO: Add default save path depending on OS
+        get_savepath_button = ttk.Button(
+            self.mainframe, command=self.get_save_file_path, text="Browse..."
+        )
+
+        self.filepath = tkinter.StringVar()
+
+        filepath_label = ttk.Label(self.mainframe, text="Save N-Trans dictionary to")
+
+        show_filepath = ttk.Entry(self.mainframe, textvariable=self.filepath, width=25)
+
+        # What N-Grams to include in csv file.
+        ngram_check_label = ttk.Label(self.mainframe, text="Choose N-grams to include")
+
+        self.deselected = 0
+        self.selected = 1
+
+        self.checked_select_all = tkinter.IntVar(value=self.selected)
+        self.checked_2_gram = tkinter.IntVar(value=self.selected)
+        self.checked_3_gram = tkinter.IntVar(value=self.selected)
+        self.checked_4_gram = tkinter.IntVar(value=self.selected)
+        self.checked_5_gram = tkinter.IntVar(value=self.selected)
+        self.checked_6_gram = tkinter.IntVar(value=self.selected)
+
+        select_all_check = ttk.Checkbutton(
+            self.mainframe,
+            text="Select all",
+            variable=self.checked_select_all,
+            command=self.select_all_ngrams,
+        )  # TODO: Add method that selects/deselects all and changes the text to reflect state.
```

Comment:
Maybe a `Button` would make more sense than a `Checkbutton`, if its text will be changing?